src/sentry/static/sentry/app/utils/tokenizeSearch.tsx

```diff
@@ -203,8 +207,19 @@ export class QueryResults {
         const token = this.tokens[i];
         const prev = this.tokens[i - 1];
         const next = this.tokens[i + 1];
-        if (isOp(token) && ['OR', 'AND'].includes(token.value)) {
-          if (prev === undefined || isOp(prev) || next === undefined || isOp(next)) {
+        if (isOp(token) && isNotParen(token.value)) {
+          if (
+            (prev === undefined || isOp(prev) || next === undefined || isOp(next)) &&
+            // Want to avoid removing `(term) OR (term)`
+            !(
+              prev !== undefined &&
+              next !== undefined &&
+              isOp(prev) &&
+              prev.value === ')' &&
+              isOp(next) &&
+              next.value === '('
+            )
+          ) {
```

Comment:
This is gonna be pseudo code but what if we did something like this

```javascript
if (isBooleanOp(token)) {
  if ((prev === undefined || isOp(prev) || next === undefined || isOp(next)) {
    if ((isParens(prev) && prev.value === ')' && isParens(next) && next.value === '(') {
      continue;
    }
    toRemove = I;
    break;
  }
}
```
Could go further and introduce `isOpenParens` and `isCloseParens` but I don't have a strong opinion there.